stack-graphs/src/stitching.rs

```diff
@@ -708,8 +725,11 @@ pub struct ForwardPartialPathStitcher {
     max_work_per_phase: usize,
     #[cfg(feature = "copious-debugging")]
     phase_number: usize,
+    should_extend: Box<ShouldExtend>,
```

Comment:
I'm not a huge fan of this being handled via a boxed function pointer, but it seems like it's only a temporary measure until we can fully remove the deprecated method.  ~Given that, I'm okay with it, as long as we put a big `TODO` comment explaining that, and proceed with removing the deprecated method (and cleaning up this field) ASAP.~ Oh hey look #184

Reply:
Hehe. I do struggle with this pattern though. Sometimes you want to plug in a little function to control behavior without adding boxes or traits etc. I think I tried using a reference to an `Fn` trait here, but that didn't work. I don't quite remember why, though.